PackageExplorer/Interop/NativeDragDrop.cs

```diff
@@ -204,68 +204,51 @@ public override void Flush()
                 _inner.Commit(0); // https://msdn.microsoft.com/en-us/library/windows/desktop/aa380320%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396
             }
 
-            public override int Read(byte[] buffer, int offset, int count)
+            public override unsafe int Read(byte[] buffer, int offset, int count)
             {
-                var ptr = Marshal.AllocHGlobal(Marshal.SizeOf<ulong>());
-                try
+                if (offset > 0)
                 {
-                    if (offset > 0)
-                    {
-                        buffer = buffer.Skip(offset).ToArray();
-                    }
+                    buffer = buffer.Skip(offset).ToArray();
+                }
 
-                    _inner.Read(buffer, count, ptr);
+                ulong lng = 0;
+                var p = &lng;
 
-                    return (int)Marshal.ReadInt64(ptr);
-                }
-                finally
-                {
-                    Marshal.FreeHGlobal(ptr);
-                }
+                _inner.Read(buffer, count, (IntPtr)p);
+
+                return (int)lng;
             }
 
-            public override long Seek(long offset, SeekOrigin origin)
+            public override unsafe long Seek(long offset, SeekOrigin origin)
             {
-                var ptr = Marshal.AllocHGlobal(Marshal.SizeOf<ulong>());
-                try
-                {
-                    _inner.Seek(offset, (int)origin, ptr);
+                long lng = 0;
+                var p = &lng;
 
-                    return Marshal.ReadInt64(ptr);
-                }
-                finally
-                {
-                    Marshal.FreeHGlobal(ptr);
-                }
+                _inner.Seek(offset, (int)origin, (IntPtr)p);
+
+                return lng;
             }
 
             public override void SetLength(long value)
             {
                 _inner.SetSize(value);
             }
 
-            public override void Write(byte[] buffer, int offset, int count)
+            public override unsafe void Write(byte[] buffer, int offset, int count)
             {
-                var ptr = Marshal.AllocHGlobal(Marshal.SizeOf<ulong>());
-                try
+                if (offset > 0)
                 {
-                    if (offset > 0)
-                    {
-                        buffer = buffer.Skip(offset).ToArray();
-                    }
+                    buffer = buffer.Skip(offset).ToArray();
+                }
 
-                    _inner.Write(buffer, count, ptr);
+                var written = 0;
```

Comment:
Should this be a `long` rather than `int`? Don't know the api that well

Reply:
In the docs I coun't find anything about the type https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.comtypes.istream.write(v=vs.110).aspx

But I thought that if the `count` is an `int` then the result can't be bigger.

Reply:
Yeah docs for Read has **`ULONG`** in bold for the return type https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.comtypes.istream.read(v=vs.110).aspx

Suppose if it works 😄

Reply:
Ah it says refer to `ISequentialStream::Write` which does say its a pointer to a `ulong` https://msdn.microsoft.com/en-us/library/windows/desktop/aa380014(v=vs.85).aspx

However as the other param is also a ulong its likely a C ulong; meaning they are all C# uint/int sizes. Same for the other methods